src/cirrus/lib/process_payload.py

```diff
@@ -36,58 +37,50 @@ class TerminalError(Exception):
     pass
 
 
-class ProcessPayload(dict):
+class ProcessPayload:
     def __init__(self, *args, set_id_if_missing=False, state_item=None, **kwargs):
         """Initialize a ProcessPayload, verify required fields, and assign an ID
 
         Args:
             state_item (Dict, optional): Dictionary of entry in StateDB.
                 Defaults to None.
         """
-        super().__init__(*args, **kwargs)
+        self.payload = Payload(*args, **kwargs)
+        self.payload.validate()
 
-        self.logger = get_task_logger(__name__, payload=self)
+        self.logger = get_task_logger(__name__, payload=self.payload)
 
-        if "process" not in self:
+        if "process" not in self.payload:
             raise ValueError(
-                "ProcessPayload must have a `process` array of process definintions",
+                "Payload must contain a 'process' array of process definitions",
             )
 
-        if not isinstance(self["process"], list) or len(self["process"]) == 0:
+        if (
+            not isinstance(self.payload["process"], list)
+            or len(self.payload["process"]) == 0
+        ):
             raise TypeError(
-                "ProcessPayload `process` must be an array "
+                "Payload 'process' field must be an array "
                 "with at least one process definition",
             )
 
-        self.process = self["process"][0]
+        self.process = self.payload.process_definition
 
-        self.features = self.get("features", [])
+        self.features = self.payload.items_as_dicts
 
-        if "id" not in self and set_id_if_missing:
+        if "id" not in self.payload and set_id_if_missing:
             self.set_id()
 
-        if "upload_options" not in self.process:
+        if "workflow" not in self.process:
             raise ValueError(
-                "ProcessPayload.process must have `upload_options` defined",
+                "Payload must contain a 'workflow' field specifying the workflow name",
             )
 
-        if "workflow" not in self.process:
+        if "workflow-" not in self.payload["id"]:
             raise ValueError(
-                "ProcessPayload.process must have `workflow` specifying workflow name",
+                f"Payload 'id' field must contain 'workflow-': {self.payload['id']}",
             )
```

Comment:
Looking at this again, would it make more sense to subclass the `Payload` class here in cirrus and put this added validation into the validation method? It seems like it would be cleaner and a clearer separation of concerns.

Setting the ID is also more specific to the `Payload` and less to this `ProcessPayload` abstraction? Maybe?

Of course setting `self.process` and `self.features` would stay here, they were simply collateral per their falling right in the middle of all this.

Anyway, something to consider. I _think_ it could be cleaner but maybe it's just another layer of indirection and would be confusing.

Reply:
I like it. It does make me question the name `ProcessPayload`, though. Our changes seem to be pushing toward a `PayloadProcessor` that has a `Payload`.

Reply:
This is done. See PR on upstream repo here: https://github.com/cirrus-geo/cirrus-geo/pull/331